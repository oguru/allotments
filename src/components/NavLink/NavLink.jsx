import {Link} from "react-router-dom";
import PropTypes from "prop-types";
import React from "react";
import styles from "./NavLink.module.scss";

const NavLink = (props) => {
   const {
      closeNav,
      isActive,
      linkText,
      path
   } = props;

   NavLink.propTypes = {
      closeNav: PropTypes.func,
      isActive: PropTypes.bool,
      linkText: PropTypes.string,
      path: PropTypes.string
   };

   const activeStyle = isActive ? "activeLink" : "";

   return (
      <div className={`${styles.navbarLink}`}>
         <Link
            className={`
              ${styles[activeStyle]}
              nav-link
              rounded`
            }
            data-test="navLink"
            onClick={closeNav}
            to={path}
         >
            {linkText}
         </Link>
      </div>
   );
};

export default NavLink;