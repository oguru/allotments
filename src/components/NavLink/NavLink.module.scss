@import "../../variables.scss";
@import "../../../node_modules/bootstrap/scss/mixins/breakpoints";
@import "../../../node_modules/bootstrap/scss/functions";
@import "../../../node_modules/bootstrap/scss/variables";

.navbarLink {
    @include transition-t;
    margin: 0.5rem;
   
    a {
        align-items: center;
        background-color: rgba(255, 255, 255, 0.05);
        border: $light-border-color 1.5px solid;
        color: rgb(0, 0, 0);
        display: flex;
        font-family: 'Fredericka the Great', cursive;
        font-size: 1.5rem;
        height: 100%;
        justify-content: flex-start;
        padding: 0.75rem 1rem;
        text-shadow: 0px 0px 0px rgb(110, 110, 110);
        transition-duration: 0.3s;

        &:active {
            background-color: none;
            border: none;
        }
    
        &:hover,
        &:focus-visible {
            background-color: $secondary-color;
            border: $light-border-color 1.5px solid;
            color: white;
            text-shadow: 0px 0px 0px rgb(180, 180, 180);
        }
    }

    .activeLink {
        border: rgb(197, 197, 197) 1.5px solid;
        border-radius: 0.25rem;
    }
}

@include media-breakpoint-up(md) {
    .navbarLink {
        margin: 0.375rem;
        min-width: 80px;
        text-align: center;
        
        a {
            border: none;
            color:rgb(214, 214, 214);
            font-size: 1.375rem;
            justify-content: center;
            padding: 0.325rem;
            text-shadow: 0px 0px 0px #b4b4b4;
            
            &:hover,
            &:focus {
                border: none;
                color: white;
                text-shadow: 0px 0px 0px #ffffff;
            }
        }
        
        .activeLink {
            border: none;
            color: white;
        }
    }
}

@include media-breakpoint-up(lg) {
    .navbarLink {
        margin: 0.5rem;
        min-width: 90px;
    }
}