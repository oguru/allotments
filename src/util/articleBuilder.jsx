import GhostImageWrapper from "../components/GhostImageWrapper/GhostImageWrapper";
import React from "react";

let count = 0;

const buildEl = (el, article) => {
   const imgCaption = el.caption ? `${"imgCaption"}` : "";

   if (el.floatImage) {
      return (
         <>
            <GhostImageWrapper
               alt={el.alt}
               classes={`
                    ${el.floatDir} 
                    ${imgCaption}`
               }
               key={`${article || "about"}${count++}`}
               src={el.floatImage}
            />
            {el.content.map(subEl => buildEl(subEl))}
         </>
      );
   }

   return (
      <React.Fragment key={count++}>
         {el.subHeading &&
             <h5>{el.subHeading}</h5>
         }
         {el.bold &&
             <p className={`
                ${"bold"} 
                ${"articleTxt"}`
             }>
                {el.bold}
             </p>
         }
         {el.li &&
             <ul className={"ulStyle"}>
                {el.li.map(li => <li key={li}>{li}</li>)}
             </ul>
         }
         {el.liText &&
             <ul className={"liText"}>
                {el.liText.map(liText => <li key={liText}>{liText}</li>)}
             </ul>
         }
         {el.imageSm && el.imageLg &&
            <GhostImageWrapper
               alt={el.alt}
               classes={`${"blockImg"} ${imgCaption}`}
               srcSet={`${el.imageSm} 300w, ${el.imageLg} 1024w`}
               src={el.imageSm}
            />
         }
         {el.splitImage &&
             <div className={`
                ${"splitImgCont"} 
                ${imgCaption}`
             }>
                <GhostImageWrapper
                   alt={el.splitImage.img1.alt}
                   classes={"splitImage"}
                   src={el.splitImage.img1.img}
                />
                <GhostImageWrapper
                   alt={el.splitImage.img2.alt}
                   classes={"splitImage"}
                   src={el.splitImage.img2.img}
                />
             </div>
         }
         {el.text &&
             <p className={`
                ${imgCaption} 
                ${"articleTxt"}`
             }>
                {el.text}
             </p>
         }
         {el.link &&
             <a
                className={"articleLink"}
                href={el.link}
                rel="noreferrer"
                target="_blank"
             >
                {el.link}
             </a>
         }
         {el.imageGrid &&
         <div className={"imageGridCont"}>
            {el.imageGrid.map(({img, text, alt}) => {
               return (
                  <div key={img}
                     className="gridBox"
                  >
                     <GhostImageWrapper classes="gridImage" src={img} alt={alt} />
                     <p className={"imgCaption articleTxt"}>{text || ""}</p>
                  </div>
               );
            })}
         </div>
         }
      </React.Fragment>
   );
};

const getContentJsx = (contentData, articles) => {
   const jsxContent = articles ?
      contentData.map(article => {
         return {
            credit: article.credit,
            id: article.id,
            mainImg: article.mainImg,
            mainImgBox: article.mainImgBox,
            mainImgAlt: article.mainImgAlt,
            path: article.path,
            title: article.title,
            initText: article.content[0].text,
            content: article.content.map(el => buildEl(el, article))
         };
      }) :
      contentData.map(el => buildEl(el));

   return jsxContent;
};

export {getContentJsx};