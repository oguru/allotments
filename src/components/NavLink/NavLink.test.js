/* eslint-disable function-paren-newline */
import {checkProps, findByTestAttr} from "../../util/utils";
import {MemoryRouter} from "react-router-dom";
import NavLink from "./NavLink";
import React from "react";
import mount from "enzyme/mount";

describe("NavLink tests", () => {
   let component;
   const testProps = {
      handleCloseNav: jest.fn(),
      linkStyle: "",
      linkText: "test text",
      path: "/testpath"
   };

   beforeEach(() => {
      component = mount(
         <MemoryRouter>
            <NavLink
               handleCloseNav={testProps.handleCloseNav}
               linkStyle={testProps.linkStyle}
               linkText={testProps.linkText}
               path={testProps.path}
            />
         </MemoryRouter>);
   });

   it("should match the snapshot", () => {
      expect(component)
         .toMatchSnapshot();
   });

   test("NavLink PropTypes check should not throw a warning", () => {
      const propsErr = checkProps(NavLink, testProps);

      expect(propsErr).toBeUndefined();
   });

   it("should gain the correct CSS style if linkStyle === 'activeLink'", () => {
      expect(findByTestAttr(component, "navLink")
         .hasClass("activeLink"))
         .toBe(false);

      component.setProps({
         children: React
            .cloneElement(
               component
                  .props()
                  .children
               , {linkStyle: "activeLink"}
            )
      });

      expect(findByTestAttr(component, "navLink")
         .hasClass("activeLink"))
         .toBe(true);
   });

   it("should call handleCloseNav when a link is clicked", () => {
      findByTestAttr(component, "navLink").simulate("click");

      expect(testProps.handleCloseNav.mock.calls.length).toBe(1);
   });

   it("should display render linkText and path props correctly", () => {
      expect(findByTestAttr(component, "navLink")
         .find(`[href="${testProps.path}"]`)
         .length)
         .toBe(1);

      expect(findByTestAttr(component, "navLink")
         .text())
         .toBe(testProps.linkText);
   });
});
