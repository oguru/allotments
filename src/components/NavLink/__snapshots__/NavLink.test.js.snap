// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink tests should match the snapshot 1`] = `ShallowWrapper {}`;
